pkg/controller/job/job_controller.go

```diff
@@ -234,7 +234,9 @@ func (jm *Controller) Run(ctx context.Context, workers int) {
 		go wait.UntilWithContext(ctx, jm.worker, time.Second)
 	}
 
-	go wait.UntilWithContext(ctx, jm.orphanWorker, time.Second)
+	for i := 0; i < workers; i++ {
+		go wait.UntilWithContext(ctx, jm.orphanWorker, time.Second)
```

Comment:
Minor nit, can you just move this invocation over to the above for-loop, no need to repeat that code.